horovod/mxnet/__init__.py

```diff
@@ -169,16 +179,25 @@ def _allreduce_grads(self):
                     grads, names = zip(*entries)
                     grouped_allreduce_(tensors=grads, average=False, name="{}:{}".format(names[0], names[-1]), priority=-i,
                                        prescale_factor=1.0 / self._gradient_predivide_factor)
+
+            if self._compression != Compression.none:
+                for i, param in enumerate(self._params):
+                    if param.grad_req != 'null':
+                        param.list_grad()[0][:] = self._compression.decompress(grads[i], ctxs[i])
```

Comment:
Using `grads[i]` and `ctx[i]` here is not quite correct. The only gradients/ctx in the `grads` and `ctx` lists are those with `param.grad_req != 'null'` already. You need to use a different index instead of `i` to access these lists.

Reply:
@romerojosh Thanks for your reminder. There is another problem in my code. When the `groups>1`, `grads` only store the last `grouped_allreduce_`'s `grads`.
I think that it can be solved by using an another list `tensors_compressed` to store all the compressed `param.list_grad`. And decompress `tensors_compressed` after `grouped_allreduce_` finish.